src/test/frontend/deploy/deployfromsettings_controller_test.js

```diff
@@ -336,4 +336,39 @@ describe('DeployFromSettings controller', () => {
       expect(ctrl.imagePullSecret).toEqual('');
     });
   });
+
+  /**
+   * Test for the name validation pattern (namePattern)
+   *
+   * The pattern matches strings that start and end with a lowercase letter
+   * and contains only lowercase letters and hyphens.
+   * Full pattern used for validation:
+   *
+   * ^[a-z]+(-[a-z]+)*$
+   */
```

Comment:
I have actually changed my test to this but not updated the test comment yet, my bad, I apologise (this back-end regex gives the same results as my shorter one, but I adopted the original one for consistency). I will update the comment of course. However, I think @cheld is suggesting that the input from this field is used in 4 different places on the back end, each with different sets of requirements. This current RegEx matches the most restrictive of the four, (only lowercase and hyphens) with the outcome that, for example, prefixes with "/" cannot be used. This is not a new restriction as names with anything other than lowercase and hyphens will currently be rejected from the backend (doesn't match the most restrictive regex)